### PR TITLE
13.0 mig partner contact in several companies

### PR DESCRIPTION
**Pull request by foliveira-odoogap**
[MIG] 13.0 partner_contact_in_several_companies

### CODE REVIEW

**Comment by rousseldenis** [comment]
@foliveira-odoogap I think this is missing migration commit, no ? Or is it WIP ?

**Comment by foliveira-odoogap** [comment]
> @foliveira-odoogap I think this is missing migration commit, no ? Or is it WIP ?

@rousseldenis no, it's not missing any migration commit

File: partner_contact_in_several_companies/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+{
+    "name": "Contacts in several partners",
+    "summary": "Allow to have one contact in several partners",
+    "version": "12.0.2.0.0",
```
Comment (rousseldenis) [inline partner_contact_in_several_companies/__manifest__.py]: @foliveira-odoogap That's make me think you forget to embed migration commit.

Comment (foliveira-odoogap) [reply partner_contact_in_several_companies/__manifest__.py]: My bad, I'll correct that as soon as possible!

Comment (foliveira-odoogap) [reply partner_contact_in_several_companies/__manifest__.py]: @rousseldenis I embed the migration commit.

File: partner_contact_in_several_companies/views/res_partner.xml [hunk 2]
```diff
@@ -1,149 +1,165 @@
...
-                        <templates>
-                            <t t-name="kanban-box">
-                                <t t-set="color" t-value="kanban_color(record.color.raw_value)"/>
-                                <div t-att-class="color + (record.title.raw_value == 1 ? ' oe_kanban_color_alert' : '') + ' oe_kanban_global_click'">
-                                    <a t-if="!read_only_mode" type="delete" class="fa fa-times pull-right"/>
-                                    <div class="o_kanban_image">
-                                        <img t-if="record.image.raw_value" t-att-src="'data:image/png;base64,'+record.image.raw_value"/>
-                                        <t t-if="!record.image.raw_value">
-                                            <img t-if="record.is_company.raw_value === true" t-att-src='_s + "/base/static/src/img/company_image.png"'/>
-                                            <img t-if="record.is_company.raw_value === false" t-att-src='_s + "/base/static/src/img/avatar.png"'/>
```
Comment (regispirard) [inline partner_contact_in_several_companies/views/res_partner.xml]: company_image.png and avatar.png where moved to /base/static/img/

**Comment by regispirard** [comment]
@foliveira-odoogap : Except for my 3 previous comments (minor changes to do), it's all good, thank you 👍

**Review by regispirard — CHANGES_REQUESTED** [review]
@foliveira-odoogap : Except for my 3 previous comments (minor changes to do), it's all good, thank you 👍

**Comment by foliveira-odoogap** [comment]
> @foliveira-odoogap : Except for my 3 previous comments (minor changes to do), it's all good, thank you
@regispirard Ok, I'll fix it as soon as I can!

**Review by HaraldPanten — CHANGES_REQUESTED** [review]
Could you check travis and commets, please?

**Review by regispirard — APPROVED** [review]
LGTM (code + functional review)

**Comment by foliveira-odoogap** [comment]
@regispirard I'm having some troubles trying to fixing those failures, could you tell me what I need to do, please?

Screenshot from 2020-09-24 16-19-15

**Comment by rousseldenis** [comment]
> @regispirard I'm having some troubles trying to fixing those failures, could you tell me what I need to do, please?
> 
> Screenshot from 2020-09-24 16-19-15

You have to install pre-commit in the repo you are  working on before comitting. ('pre-commit install').

You have error here:

image

**Comment by foliveira-odoogap** [comment]
@rousseldenis everything's ok to merge =)

**Comment by regispirard** [comment]
> @rousseldenis everything's ok to merge =)

Good job ;-)

**Comment by rousseldenis** [comment]
Cool, could you squash your fixes commits into the migration one ? Thanks

**Comment by foliveira-odoogap** [comment]
> Cool, could you squash your fixes commits into the migration one ? Thanks

By squash, you mean doing rebase with my branch 13.0?

**Comment by foliveira-odoogap** [comment]
@rousseldenis can you confirm if everything's ok to proceed?

**Review by HaraldPanten — APPROVED** [review]

**Comment by NL66278** [comment]
@foliveira-odoogap Squashing your commits basically means that all the commits you made for the migration are merged into one commit. You have to do an interactive rebase from the commit just before your migration commit: $ git rebase -i <commit to rebase from>. You might use $ git rebase -i HEAD~5 to get the last 5 commits. You then get an editor screen with each commit on one line. You can mark all the commits after the first as a fixup, by replacing the commit hash with an 'f'. May sound complicated, but is actually quite easy.

**Comment by dreispt** [comment]
@foliveira-odoogap Can you please squash commit, as per https://github.com/OCA/partner-contact/pull/946#issuecomment-712168420
Let me know if you need help

**Comment by foliveira-odoogap** [comment]
> @foliveira-odoogap Can you please squash commit, as per [#946 (comment)](https://github.com/OCA/partner-contact/pull/946#issuecomment-712168420)
> Let me know if you need help

Hi @dreispt, maybe it would be better for you to help me.

**Comment by foliveira-odoogap** [comment]
> > @foliveira-odoogap Can you please squash commit, as per [#946 (comment)](https://github.com/OCA/partner-contact/pull/946#issuecomment-712168420)
> > Let me know if you need help
> 
> Hi @dreispt, maybe it would be better for you to help me.

@dreispt can you help me please?

**Comment by rousseldenis** [comment]
@foliveira-odoogap What do you need ?

**Comment by foliveira-odoogap** [comment]
> @foliveira-odoogap What do you need ?

I don't know what you mean by squashing my commits, can tell me how you do that please?

**Comment by rousseldenis** [comment]
> > @foliveira-odoogap What do you need ?
> 
> I don't know what you mean by squashing my commits, can tell me how you do that please?

When developing, it's common to commit step by step, adding or removing features.

But some commits have meaning (enclosed modifications have sense) and some could have not.

So, git have feature to 'merge' some commits in the stack together.

To do so, see: https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git

It's an interactive rebase. You choose how many commits to rebase and then modify the operation on each of them (pick/fixup/drop/rename...).

In your case, git rebase -i HEAD~6, fixup for the five last ones and pick for the Migration one.

**Comment by foliveira-odoogap** [comment]
> > > @foliveira-odoogap What do you need ?
> > 
> > 
> > I don't know what you mean by squashing my commits, can tell me how you do that please?
> 
> When developing, it's common to commit step by step, adding or removing features.
> 
> But some commits have meaning (enclosed modifications have sense) and some could have not.
> 
> So, git have feature to 'merge' some commits in the stack together.
> 
> To do so, see: https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git
> 
> It's an interactive rebase. You choose how many commits to rebase and then modify the operation on each of them (pick/fixup/drop/rename...).
> 
> In your case, git rebase -i HEAD~6, fixup for the five last ones and pick for the Migration one.

Ok, I squashes my commits and pushed them, now you have to merge right?

**Comment by rousseldenis** [comment]
Indeed, as soon as you have pushed and tests green, I'll do it

**Comment by remihb** [comment]
> In your case, git rebase -i HEAD~6, fixup for the five last ones and pick for the Migration one.

Yes @foliveira-odoogap please do as @rousseldenis explains.
I cloned your code and still have your 6 migration commits. Maybe you forget to push ?

**Comment by HaraldPanten** [comment]
@foliveira-odoogap any news about this?

**Comment by remihb** [comment]
something went wrong !

**Pull request merged**